packages/grid/x-data-grid/src/hooks/features/export/serializers/csvSerializer.ts

```diff
@@ -111,10 +153,43 @@ export function buildCSV(options: BuildCSVOptions): string {
     return CSVBody;
   }
 
-  const CSVHead = `${columns
-    .filter((column) => column.field !== GRID_CHECKBOX_SELECTION_COL_DEF.field)
-    .map((column) => sanitizeCellValue(column.headerName || column.field, delimiterCharacter))
-    .join(delimiterCharacter)}\r\n`;
+  const filteredColumns = columns.filter(
+    (column) => column.field !== GRID_CHECKBOX_SELECTION_COL_DEF.field,
+  );
+
+  const headerRows: CSVRow[] = [];
+
+  if (includeColumnGroupsHeaders) {
+    const columnGroupLookup = apiRef.current.unstable_getAllGroupDetails();
+
+    let maxColumnGroupsDepth = 0;
+    const columnGroupPathsLookup = filteredColumns.reduce<
+      Record<GridStateColDef['field'], GridColumnGroup['groupId'][]>
+    >((acc, column) => {
+      const columnGroupPath = apiRef.current.unstable_getColumnGroupPath(column.field);
+      acc[column.field] = columnGroupPath;
+      maxColumnGroupsDepth = Math.max(maxColumnGroupsDepth, columnGroupPath.length);
+      return acc;
+    }, {});
+
+    for (let i = 0; i < maxColumnGroupsDepth; i += 1) {
```

Comment:
Since it's not possible to merge cells in CSV, we can have 2 approaches to reflecting the column groups.
Consider this example:

<img width="571" src="https://github.com/mui/mui-x/assets/13808724/3514b69b-ea29-426b-a595-06addfd7ea63">

1. Duplicate column group headers:
  	<img width="458" src="https://github.com/mui/mui-x/assets/13808724/4c149283-e053-465f-82a2-a636c205a417">
  	
  	This looks verbose, but it's clear that the "Age" column belongs to the "Basic info" group and does not belong to the "Full name" group
  
2. Avoid duplicating column group headers
	<img width="393" src="https://github.com/mui/mui-x/assets/13808724/4e31ad4e-4850-4268-8544-cfb33b695a72">
	
	It looks less verbose, but in this case, the group structure is unclear. Does the "Age" column belong to the "Full name" and "Basic info" groups?


I prefer option 1 because it's more clear.
What is your opinion?

Reply:
1

Reply:
Yes, no. 1 is more clear to me too. 👍